### PR TITLE
feat(storagenode): add gRPC error codes to the admin service

### PR DESCRIPTION
**Pull request by ijsong**
### What this PR does

This patch adds gRPC error codes to the RPCs of the admin service in the storage node.

### Which issue(s) this PR resolves

Updates #312

### CODE REVIEW

**Comment by ijsong** [comment]
Current dependencies on/for this PR:
* main
  * **PR #318** <a href="https://app.graphite.dev/github/pr/kakao/varlog/318" target="_blank">Graphite</a> 
    * **PR #328** <a href="https://app.graphite.dev/github/pr/kakao/varlog/328" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/kakao/varlog/328?utm_source=stack-comment).

File: internal/storagenode/admin_server.go [hunk 1]
```diff
@@ -60,7 +124,19 @@ func (as *adminServer) Sync(ctx context.Context, req *snpb.SyncRequest) (*snpb.S
 			LogStreamID: req.LogStreamID,
 		},
 	})
-	return &snpb.SyncResponse{Status: syncStatus}, err
+	if err != nil {
+		var code codes.Code
+		switch err {
```
Comment (hungryjang) [inline internal/storagenode/admin_server.go:129]: How about integrating with the L103 code?

Comment (ijsong) [reply internal/storagenode/admin_server.go:129]: Are you suggesting refactoring, such as extracting the common switch block?

Comment (ijsong) [reply internal/storagenode/admin_server.go:129]: If you mean extracting out and integrating the common switch block, I want instead to propose another approach, such as intercepting common errors like ErrClosed and ErrNotExist.

Comment (hungryjang) [reply internal/storagenode/admin_server.go:129]: I mean to create a separate function that replaces snerrors with grpc error code. There is no need to do that if 1:1 matching is not possible.

**Review by hungryjang — APPROVED** [review]

**Pull request merged**